src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenLocalFunctionTests.cs

```diff
@@ -90,24 +182,24 @@ .maxstack  2
   IL_0010:  ldarg.0
   IL_0011:  ldfld      ""int C._x""
   IL_0016:  call       ""void System.Console.WriteLine(int)""
-  IL_001b:  ldloca.s   V_0
-  IL_001d:  call       ""void C.<M>g__L12_0(ref C.<>c__DisplayClass2_0)""
-  IL_0022:  ldarg.0
-  IL_0023:  ldfld      ""int C._x""
-  IL_0028:  call       ""void System.Console.WriteLine(int)""
-  IL_002d:  ret
+  IL_001b:  ldarg.0
```

Comment:
Yup, this is a result of a less aggressive optimization. What's going on here is we only eliminate the `this` closure if it captures *only* `this`. But both `x` and `this` are captured in the same scope, so the environment contains both `x` and `this`, thus the optimization doesn't apply. I haven't found a really simple optimization strategy that works for this case, so I didn't do anything extra here.

If you have any suggestions I could incorporate them, though.